modular_skyrat/modules/paycheck_rations/code/quirk.dm

```diff
@@ -59,21 +54,18 @@
 		return
 
 	var/obj/item/created_ticket
-	for(var/obj/card in bank_cards)
-		// We want to only make one ticket pr account per payday
-		if(created_ticket)
-			continue
+	for(var/obj/card in account.bank_cards)
+		// We want to only make one ticket per account per payday
```

Comment:
This comment seems incorrect now- also, doesn't this mean there's now a ticket for every account? Before this change this code either inserted succesfully, in which case it would have hit the continue every time, or it would fail, in which case it returned. In effect only the first bank card was considered, rendering a loop wholly unnecessary

Reply:
It still makes one ticket per account. The comment was incorrect before - you kept parsing through bank cards even though the account already had a ticket, so you could've broken out. Yes, whoever coded this before me didn't know you can break out of loops and did this instead.

I honestly don't know why it doesn't just index into the first slot of the list but the for loop automatically typechecks for us so I guess it's safer?

Reply:
It still makes one ticket per account. The comment was incorrect before - you kept parsing through bank cards even though the account already had a ticket, so you could've broken out. Yes, whoever coded this before me didn't know you can break out of loops and did this instead.

I honestly don't know why it doesn't just index into the first slot of the list but the for loop automatically typechecks for us so I guess it's safer?